src/settingsview.cpp

```diff
@@ -62,6 +62,21 @@ SettingsView::SettingsView(QWidget *parent)
     ui->monitorHelp->setToolTip(gt("monitor-directory-tooltip"));
     ui->moveToTrashLabel->setText(gt("move-files-to-trash"));
     ui->reopenTabLabel->setText(gt("open-previous-tabs-at-startup"));
+    if(isPortableMode()) {
+        ui->browseButton->setEnabled(false); 
+        ui->resetButton->setEnabled(false); 
+        ui->monitorBrowse->setEnabled(false); 
+        ui->monitorClear->setEnabled(false); 
+        QString disabledStyle = "color: gray;";
+        ui->browseButton->setStyleSheet(disabledStyle);
+        ui->resetButton->setStyleSheet(disabledStyle);
+        ui->monitorBrowse->setStyleSheet(disabledStyle);
+        ui->monitorClear->setStyleSheet(disabledStyle);
+        ui->browseButton->setToolTip(gt("portable-disabled-tooltip"));
+        ui->resetButton->setToolTip(gt("portable-disabled-tooltip"));
+        ui->monitorBrowse->setToolTip(gt("portable-disabled-tooltip"));
+        ui->monitorClear->setToolTip(gt("portable-disabled-tooltip"));
```

Comment:
Please rewrite this fragment in a shorter way after introducing the following helper function in an unnamed namespace

```c++
void disableInPortableMode(QWidget* w)
{
        w->setEnabled(false); 
        QString disabledStyle = "color: gray;";
        w->setStyleSheet(disabledStyle);
        w->setToolTip(gt("portable-disabled-tooltip"));
}
```